ui/common/src/commonMain/kotlin/social/androiddev/common/utils/AsyncImage.kt

```diff
@@ -9,56 +9,29 @@
  */
 package social.androiddev.common.utils
 
-import androidx.compose.foundation.Image
-import androidx.compose.foundation.layout.Spacer
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.produceState
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.graphics.painter.Painter
 import androidx.compose.ui.layout.ContentScale
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.withContext
-import social.androiddev.common.network.util.runCatchingIgnoreCancelled
+import io.kamel.image.KamelImage
+import io.kamel.image.lazyPainterResource
 
 /**
  * Use this helper until we switch to a image loading library which supports multiplatform
  */
 @Composable
-fun <T> AsyncImage(
+fun AsyncImage(
     contentDescription: String,
     modifier: Modifier = Modifier,
     alignment: Alignment = Alignment.Center,
     contentScale: ContentScale = ContentScale.Fit,
-    load: suspend () -> T,
-    painterFor: @Composable (T) -> Painter,
+    url:String
```

Comment:
Similar to how coil for compose works: it would be nice to never have to differentiate at call site if we should use AsyncImage or built in Image, and be able to use a single Image compose wrapper for rendering url or a drawable. 

Can we change this to `Any` if KamelImage supports it?

Reply:
yup yup it seems to delegate to same chain